src/main.rs

```diff
@@ -22,53 +25,71 @@ use opengl_graphics::{
     Gl,
 };
 
-use std::env;
-use std::io::{Read, BufReader};
+use std::io::Read;
 use std::fs::File;
 use std::path::Path;
 use input::Button;
 
+use docopt::Docopt;
+
 use chip8_vm::vm::Vm;
 
 const TITLE: &'static str = "Chip8";
 const BEEP_TITLE: &'static str = "♬ Chip8 ♬";
 
-const INTRO_ROM: &'static [u8] = include_bytes!("intro/intro.ch8");
+const USAGE: &'static str = "
+CHIP-8 user interface.
 
-fn main() {
-    env_logger::init().unwrap();
+Usage:
+  chip8_ui [--] <rom> | -
+  chip8_ui -h | --help
+  chip8_ui -V | --version
 
-    let mut vm = Vm::new();
+Options:
+  -h, --help     Show this screen.
+  -V, --version  Show version.
+";
 
-    let mut rom: Option<File> = None;
+#[derive(RustcDecodable, Debug)]
+struct Args {
+    arg_rom: String,
+}
 
-    if let Some(rom_path) = env::args().skip(1).next() {
-        if let Ok(f) = File::open(&Path::new(&rom_path)) {
-            rom = Some(f);
-        } else {
-            error!("Could not open ROM {}", rom_path);
-        }
+fn create_vm<R>(rom: &mut R) -> Vm
+    where R: Read
+{
+    let mut vm = Vm::new();
+    match vm.load_rom(rom) {
+        Ok(_) => vm,
+        Err(e) => panic!("Error loading ROM: {}", e)
```

Comment:
This needs more sophisticated error handling since it's an abuse of `panic!` >> `TODO`.